browser-extension/mv3/src/service-worker/services/tabService.ts

```diff
@@ -68,6 +70,10 @@ class TabService {
     chrome.webNavigation.onCommitted.addListener((navigatedTabData) => {
       if (navigatedTabData.frameId === 0) {
         this.resetPageData(navigatedTabData.tabId);
+        const isBlocked = isUrlInBlockList(navigatedTabData.url);
+        if (isBlocked) {
+          extensionIconManager.markExtensionBlocked(navigatedTabData.tabId);
+        }
```

Comment:
This should be in extensionIconManager `chrome.tabs.onUpdated` as this introduces cirular loop